packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts

```diff
@@ -222,13 +222,30 @@ export const useFieldState = <
 
     const activeSection = state.sections[selectedSectionIndexes.startIndex];
     const activeDateManager = fieldValueManager.getActiveDateManager(utils, state, activeSection);
+    const activeDateSections = fieldValueManager.getActiveDateSections(
+      state.sections,
+      activeSection,
+    );
+
+    const isOnlyNonEmptySection = activeDateSections.every((section) => {
```

Comment:
For me `isOnlyNonEmptySection` means the field only contains non-empty sections

```suggestion
    const otherSectionsAreEmpty = activeDateSections.every((section) => {
```

```suggestion
    const isTheOnlyNonEmptySection = activeDateSections.every((section) => {
```